### PR TITLE
Extract Request and Result classes from NetSuite::Client

### PR DESCRIPTION
**Pull request by base10**
NetSuite::Client is getting a bit large and with retrieving fields
from NetSuite and having to build yet another object, it makes sense to
extract these now.

* In support of PR #162
* For: https://trello.com/c/mThbJz4M/

### CODE REVIEW

**Comment by base10** [comment]
@drapergeek Per discussion, extracted this outside the context of #162. Tried flipping the order on this and seemed like it was going to work better to get this out of the way first. Review appreciated.

File: app/models/net_suite/client.rb [hunk 1]
```diff
@@ -69,85 +70,12 @@ def subsidiaries
...
-        json.each(&block)
-      end
-
-      private
-
-      def json
-        @json ||= JSON.parse(@response)
-      end
+    def request
+      @request || Request.new(
```
Comment (drapergeek) [inline app/models/net_suite/client.rb]: `||=` maybe?

Comment (base10) [inline app/models/net_suite/client.rb]: :disappointed: Fixed in 4498b7d

**Comment by drapergeek** [comment]
LGTM

**Comment by base10** [comment]
\o/ Thanks.

**Pull request merged**